Steer/DigitizerWorkflow/src/CTPDigitizerSpec.cxx

```diff
@@ -0,0 +1,79 @@
+//
+// Created by rl on 3/17/21.
+//
+
+#include "CTPDigitizerSpec.h"
+#include "Framework/ControlService.h"
+#include "DetectorsBase/BaseDPLDigitizer.h"
+#include "DataFormatsCTP/Digits.h"
+#include "Steer/HitProcessingManager.h" // for DigitizationContext
+#include "CTPSimulation/Digitizer.h"
+
+#include <TChain.h>
+#include <TStopwatch.h>
+
+using namespace o2::framework;
+namespace o2
+{
+namespace ctp
+{
+class CTPDPLDigitizerTask : public o2::base::BaseDPLDigitizer
+{
+  using GRP = o2::parameters::GRPObject;
+
+ public:
+  CTPDPLDigitizerTask() : o2::base::BaseDPLDigitizer(), mDigitizer() {}
+  ~CTPDPLDigitizerTask() override = default;
+  void run(framework::ProcessingContext& pc)
+  {
+    if (mFinished) {
+      return;
+    }
+    // read collision context from input
+    auto context = pc.inputs().get<o2::steer::DigitizationContext*>("collisioncontext");
+    context->initSimChains(o2::detectors::DetID::CTP, mSimChains);
```

Comment:
this is a mechanism to read detector hits, which don't exist for the CTP, instead you should use `triggerInput` messages from relevant detectors.

Reply:
yes, but this is first exercise bypassing triggerInputs.

Reply:
ok, I understand that it is not functional yet, but you don't need detector hits.

Reply:
ok, I gave up example without Trigger Inputs. I am trying to do the version with T0 and V0 inputs. Please, have a look.